numpy/core/tests/test_numerictypes.py

```diff
@@ -87,10 +87,8 @@ def normalize_descr(descr):
             else:
                 nitem = (item[0], dtype)
             out.append(nitem)
-        elif isinstance(item[1], list):
-            l = []
-            for j in normalize_descr(item[1]):
-                l.append(j)
+        elif isinstance(dtype, list):
+            l = normalize_descr(dtype)
```

Comment:
Are you sure this is equivalent? Previously we iterated through each element of `dtype` -- now we don't.

Reply:
Nope, we iterated through each item of `descr`. `dtype` is the name being used for the elements within the `descr`, set on line 77.